conformance/lang/expressions/logical-expr/logical_or_expr.balt

```diff
@@ -0,0 +1,853 @@
+    io:println(4 < 2 || false); // @output false
+    io:println(4 < 3 || 4 < 3); // @output false
+
+    io:println(true || 4 >= 2); // @output true
+    io:println(4 >= 2 || true); // @output true
+    io:println(false || 4 >= 2); // @output true
+    io:println(4 >= 2 || false); // @output true
+    io:println(4 >= 3 || 4 >= 3); // @output true
+
+    io:println(true || 4 <= 2); // @output true
+    io:println(4 <= 2 || true); // @output true
+    io:println(false || 4 <= 2); // @output false
+    io:println(4 <= 2 || false); // @output false
+    io:println(4 <= 3 || 4 <= 3); // @output false
+
+    io:println(4 < 2 || 4 > 2); // @output true
+    io:println(4 >= 2 || 4 <= 3); // @output true
+    io:println(4 > 2 || 4 <= 2); // @output true
+    io:println(4 >= 2 || 4 < 2); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression using is expression.
+Labels: logical-expr, is-expr
+
+function init() {
+    io:println(true || 2 is int); // @output true
+    io:println(2 is int || true); // @output true
+    io:println(false || 2 is int); // @output true
+    io:println(2 is int || false); // @output true
+    io:println(2 is int || 2 is int); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression using equality expression.
+Labels: logical-expr, equality, exact-equality
+
+function init() {
+    io:println(true || 2 == 3); // @output true
+    io:println(3 == 3 || true); // @output true
+    io:println(false || 3 == 3); // @output true
+    io:println(2 == 3 || false); // @output false
+    io:println(2 == 3 || 3 == 3); // @output true
+
+    io:println(true || 2 != 3); // @output true
+    io:println(3 != 3 || true); // @output true
+    io:println(false || 3 != 3); // @output false
+    io:println(2 != 3 || false); // @output true
+    io:println(2 != 3 || 3 != 3); // @output true
+
+    io:println(true || 2 === 3); // @output true
+    io:println(3 === 3 || true); // @output true
+    io:println(false || 3 === 3); // @output true
+    io:println(2 === 3 || false); // @output false
+    io:println(2 === 3 || 3 === 3); // @output true
+
+    io:println(true || 2 !== 3); // @output true
+    io:println(3 !== 3 || true); // @output true
+    io:println(false || 3 !== 3); // @output false
+    io:println(2 !== 3 || false); // @output true
+    io:println(2 !== 3 || 3 !== 3); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression with logical expression.
+Labels: logical-expr
+
+function init() {
+    io:println(true || true || false); // @output true
+    io:println(true || false || true); // @output true
+    io:println(false || true || false); // @output true
+    io:println(true || false || false); // @output true
+    io:println(true || false || true || false); // @output true
+}
+
+Test-Case: output
+Description: Expression evaluation order when lhs expression evaluates to false and rhs expression evaluates to true in a logical or expression.
```

Comment:
Better to elaborate what the expected behaviour is based on the spec.

Reply:
[94d5da7](https://github.com/ballerina-platform/ballerina-spec/pull/1113/commits/94d5da7d6a3c741d8d2a8cb2d22ac6910b20f776)